### DISCUSSION TITLE
resource-path for pdf class files

### DISCUSSION THREAD

**Discussion by verbalins** [quarto-dev/quarto-cli#6526] [Q&A] [answered]
### Description

Hello quarto community!

I'm developing a format extension where one of the output formats is PDF. 

Currently I'm using `format-resources` to specify the files needed for PDF compilation. This copies the files specified to the root directory and compiles as I want it to. 

However, I saw the `resource-path` option and thought I could use that to set a path reference to the supplementary files (.cls, .dbx, .clo, and a few logos). I thought the option would add it to the compiler, similar to how ensure_path('TEXINPUTS') works in .latexmkrc.

Is this not possible, or am I just doing it wrong?

**Comment by dragonstyle** [comment, marked as answer]
The `resource-path` is used by Pandoc itself when it resolves resources (not by the underlying TeX compiler). We do support `latex-input-paths` as an option to add paths to search when compiling LaTeX, but we generally don't recommend that extension developers use them for two reasons:

1) Extensions may be installed in a namespaced folder, or even nested within another extension. Further, they can be installed beside a document, or at the project root. All of this makes forming a path to the `_extensions` subfolder difficult and unlikely to work consistently.

2) We prefer copying the resources next to the input, so that if the user experiences a TeX error, they are left in a state where they could manually run the TeX compile command to aide debugging. If we inject special paths to resources, the user would then also need to do this in order to run a command like `xelatex` manually on the intermediary `.tex` file. Using format resources means that this debugging scenario will work without the user need need to alter their tex input paths.

HTH!

**Reply by verbalins** [reply]
Thank you for your detailed answer, that clears things up for me!
